packages/apputils-extension/src/index.ts

```diff
@@ -195,12 +217,11 @@ const themes: JupyterLabPlugin<IThemeManager> = {
       }
     });
 
-    // If we have a main menu, add the theme manager
-    // to the settings menu.
+    // If we have a main menu, add the theme manager to the settings menu.
     if (mainMenu) {
       const themeMenu = new Menu({ commands });
       themeMenu.title.label = 'JupyterLab Theme';
-      manager.ready.then(() => {
```

Comment:
I know you talked about this earlier, but can you go over again why you removed the `ready` promise from the theme manager?

Reply:
The theme manager's `ready` promise waited until the application started before themes were ready to render. That meant that pre-startup error conditions like loading for too long or needing to change workspace would render un-styled.